src/kOS/AddOns/KerbalAlarmClock/KACAlarmWrapper.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Linq;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Utilities;
+using kOS.AddOns.KerbalAlarmClock;
+
+namespace kOS.Suffixed
+{
+    public class KACAlarmWrapper : Structure
+    {
+        private KACWrapper.KACAPI.KACAlarm alarm;
+        private SharedObjects shared;
+
+        public KACAlarmWrapper(KACWrapper.KACAPI.KACAlarm init, SharedObjects shared) 
+        { 
+            alarm = init;
+            this.shared = shared;
+            InitializeSuffixes();
+        }
+        public KACAlarmWrapper(String alarmID, SharedObjects shared) 
+        {
+            alarm = KACWrapper.KAC.Alarms.First(z=>z.ID==alarmID);
+            this.shared = shared;
+            InitializeSuffixes();
+        }
+        private void InitializeSuffixes()
+        {
+            AddSuffix("ID", new Suffix<string>(() => alarm.ID));
+            AddSuffix("NAME", new SetSuffix<string>(() => alarm.Name, value => alarm.Name = value));
+
+            AddSuffix("VESSELID", new SetSuffix<string>(() => alarm.VesselID, value => alarm.VesselID = value));
+            AddSuffix("VESSEL", new SetSuffix<VesselTarget>(() => getVesselByID(alarm.VesselID), value => alarm.VesselID = value.Vessel.id.ToString()));
+
+            AddSuffix("NOTES", new SetSuffix<string>(() => alarm.Name, value => alarm.Name = value));
+
+            AddSuffix("ALARMACTION", new SetSuffix<string> (alarm.AlarmAction.ToString, SetAlarmAction));
+
+            AddSuffix("ALARMTYPE", new Suffix<string> (alarm.AlarmType.ToString));

```

Comment:
many of the suffixes that are prefixed with "ALARM" could be removed. you are dealing with an alarm structure so we can make them terser without loosing meaning

SET ALARM TO ADDALARM("RAW", time:seconds  + 20, "FOO", "BAR").

PRINT ALARM:ALARMTYPE.

could be 

PRINT ALARM:TYPE.